src/components/FocusBoundary.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+function isChildOf(target, child) {
+  if (!child) {
+    return false;
+  }
+
+  return target === child.parentElement
+    ? true
+    : isChildOf(target, child.parentElement);
+}
+
+export const FocusBoundary = ({ children, onChange, onEnter, onExit }) => {
+  const [ref, setRef] = React.useState();
+  const boundaryContainsFocus = React.useRef();
+
+  React.useEffect(() => {
+    if (ref) {
+      const focusListener = (e) => {
+        if (!boundaryContainsFocus.current) {
+          onChange && onChange(true);
+          onEnter && onEnter(e);
+          boundaryContainsFocus.current = true;
+        }
+      };
+      const blurListener = (e) => {
+        if (!isChildOf(ref, e.relatedTarget)) {
+          if (onChange) {
+            e.relatedTarget && e.relatedTarget.focus({ preventScroll: true });
```

Comment:
This is here to fix a scroll animation bug when used in the sidebar. Without it, the following happens:

- focus leaves the boundary and jumps to `e.relatedTarget`
- browser auto-scrolls horizontally to make the new target visible in the viewport
- `FocusBoundary` calls the `onChange` and/or `onExit` callbacks
- sidebar transition happens, scrolling the content 40% to the left
- content is only partially visible, half of the screen is blank

By preventing the browser auto-scroll, it works as expected. In future we might need to move this into the callback itself, but for now this is a fair bit cleaner and still works!